Simplenote/Classes/SPBlurEffectView.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+import UIKit
+
+
+// MARK: - SPBlurEffectView: Reacts automatically to UserInterfaceStyle Changes
+//
+@objc
+class SPBlurEffectView: UIVisualEffectView {
+
+    /// Blur's TintView
+    ///
+    private let tintView: UIView = {
+        let output = UIView()
+        output.autoresizingMask = [.flexibleWidth, .flexibleHeight];
+        return output
+    }()
+
+    /// Closure that's expected to return the TintColor we want. Runs on `iOS <13` everytime the active Theme is updated.
+    ///
+    var tintColorClosure: (() -> UIColor)? {
+        didSet {
+            refreshTintColor()
+        }
+    }
+
+
+    // MARK: - Initializers
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    init() {
+        super.init(effect: UIBlurEffect.simplenoteBlurEffect)
+        setupTintView()
+        startListeningToNotifications()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(effect: UIBlurEffect.simplenoteBlurEffect)
+        setupTintView()
+        startListeningToNotifications()
+    }
```

Comment:
A little nitpicky of me, but maybe factor out the setup and notification calls into a common method each init can call?

Reply:
@aerych can you help me understand if you are recommending this instead of…

```swift
init() {
	…
}

required init?(code: NSCode) {
	init();
}
```

also, is what I'm asking even possible?

Reply:
As currently written, `init` and `init(coder:)` both call `setupTintView` and `startListeningToNotifications`. Basically, the work is identical apart from the call to `super`. Its a trivial amount of duplication, so not really anything to fuss over, but if we wanted to be a smidge nicer we could have a single method to do the common work that each init method could call.  Something like...

```
init() {
    super.init(effect: UIBlurEffect.simplenoteBlurEffect)
    performInitialSetup()
}

required init?(coder: NSCoder) {
    super.init(effect: UIBlurEffect.simplenoteBlurEffect)
    performInitialSetup()
}

func performInitialSetup() {
    setupTintView()
    startListeningToNotifications()
}
```

Alternatively, and maybe nicer still, but I'm skeptical if it would compile...

```
init() {
    init(effect: UIBlurEffect.simplenoteBlurEffect)
}

required init?(coder: NSCoder) {
    init(effect: UIBlurEffect.simplenoteBlurEffect)
}

override init(effect: UIBlurEffect) {
    super.init(effect: effect)
    setupTintView()
    startListeningToNotifications()
}
```

Reply:
yeah I was asking more about leaving `init()` the same as it is and calling `init()` from `init(coder: NSCoder)` instead of creating a third method that both call

why go through the extra work? is it necessary because of Framework? or Swift?

Reply:
Actually it did compile! Just had to add the `convenience` keyword.

```
    convenience init() {
        self.init(effect: .simplenoteBlurEffect)
    }

    convenience required init?(coder: NSCoder) {
        self.init(effect: .simplenoteBlurEffect)
    }

    init(effect: UIBlurEffect) {
        super.init(effect: effect)
        setupTintView()
        startListeningToNotifications()
    }
```

Thank you @aerych!!!

Reply:
@dmsnell oh forgot to answer! `init(coder:` sometimes is required by the superclass, and enforced by the compiler. Not adding it just throws a build error!

Reply:
TIL you can mark a `required` initializer as `convenience`.  That makes zero sense to me but 🤷‍♂

Reply:
I didn't know it either... but the compiler is happy about it (wooo!!)